modules/concept-docs/pages/n1ql-query.adoc

```diff
@@ -14,6 +14,47 @@ Below, we fill in some of the gaps between reference and rolling-up-your-sleeves
 with discussion of a few areas of the Query service where more background knowledge will help you to better program your application.
 
 
+== Prepared Statements for Query Optimization
+
+When a N1QL query string is sent to the server, the server will inspect the string and parse it, planning which indexes to query.
+Once this is done, it generates a _query plan_.
+The computation for the plan adds some additional processing time and overhead for the query.
+
+Often-used queries can be _prepared_ so that its _plan_ is generated only once.
+Subsequent queries using the same query string will use the pre-generated plan instead, saving on the overhead and processing of the plan each time.
+
+You can indicate to the SDK that a given query should be optimized in the above fashion.
+When an SDK sees that a query should be optimized, it will internally prepare the statement and store the plan in an internal cache.
+When issuing the query again, the SDK will check to see if a plan exists in its cache, and will send the plan to the server.
+
+To indicate that an SDK should optimize a query, the `Adhoc` parameter should be set to false.
```

Comment:
I feel like this paragraph is slightly misleading, it defaults to false so queries use prepared statements by default. The wording here makes me think that I have to disable something to use prepared statements when in fact you have to enable AdHoc to not use them.

Reply:
Ah, is it specced for all of the SDKs like this?